frontend/reducers/document.js

```diff
@@ -1,40 +1,96 @@
 'use strict'
 
 import * as Constants from 'lib/constants'
+import * as LocalStorage from 'lib/local-storage'
 import * as Types from 'actions/actionTypes'
 
 const initialState = {
+  dirty: false,
+  loadedFromLocalStorage: false,
   local: null,
   peers: null,
   remote: null,
   remoteStatus: Constants.STATUS_IDLE,
   validationErrors: {}
 }
 
+// Boolean fields are a bit special. Any required field that hasn't
+// been touched by the user in the UI should generate a validation error,
+// except for Boolean fields, as these don't have a "neutral" state.
+// You can start editing a document and see a Boolean field set to
+// "false" and that may be exactly how you want it, but at that point the
+// field doesn't exist in the document store yet because there wasn't a
+// change event. To get around this, when we create our local copy of
+// the document we check for any Boolean fields in the collection
+// that aren't in the document object and set those to `false`.
```

Comment:
I don't understand how this helps us with the problem addressed in the comment. If you have a Boolean field that's required, you can't save a document without including it in the document body. For that to happen, we need to make sure we append that field to the document even if the user hasn't interacted with the UI element, which is precisely what this piece of logic is doing for Boolean fields.